### PR TITLE
Configure Application Gateway custom probe

### LINKED ISSUE #201: Configure Application Gateway Custom probe
Application Gateway custom probes now support the ability to provide multiple HTTP status codes that should be taken as healthy response for the backend pool. Previously, a healthy response was only considered when the response code was in the 200-399 range.

A custom health probe should be configured to allow 401 responses when X-Pack Security is enabled, which would remove the need to allow anonymous access to the base path `/`.

### PR DESCRIPTION
**Pull request by russcam**
This PR configures a custom probe for Application Gateway.
Custom probes now support status codes other than 200-399, so
configure the probe to also consider a 401 response as healthy.
This removes the need to configure anonymous access to the cluster
base path.

Closes #201

### CODE REVIEW

**Comment by russcam** [comment]
Run `npm run links -- --branch master` before merging

**Comment by russcam** [comment]
Opened https://github.com/MicrosoftDocs/azure-docs/issues/11805 as custom probe status code matching does not show correctly in the portal, although Application Gateway appears to honour them.

File: src/mainTemplate.json [hunk 1]
```diff
@@ -932,18 +932,8 @@
...
-        3,
-        4,
-        5,
-        6,
-        7,
-        8,
-        9,
-        10
-      ],
+      "minValue": 1,
```
Comment (Mpdreamz) [inline src/mainTemplate.json]: 👍

File: src/scripts/elasticsearch-ubuntu-install.sh [hunk 2]
```diff
@@ -1028,21 +1009,6 @@ configure_elasticsearch_yaml()
         echo "xpack.security.enabled: true" >> $ES_CONF
     fi
 
-    # Configure Anonymous access
```
Comment (Mpdreamz) [inline src/scripts/elasticsearch-ubuntu-install.sh]: ❤️  that we can :boom: this now

**Pull request merged**